plugins/catalog/src/plugin.ts

```diff
@@ -249,3 +254,12 @@ export const RelatedEntitiesCard: <T extends Entity>(
     },
   }),
 );
+
+/** @public */
+export const CatalogSearchResultListItemExtension = catalogPlugin.provide(
+  createSearchResultListItemExtension<CatalogSearchResultListItemProps>({
+    name: 'CatalogSearchResultListItem',
+    component: CatalogSearchResultListItem,
```

Comment:
Would it make sense to deprecate the old component?

Reply:
I'm not sure if it makes sense to deprecate because the extensions only return null if they are used just as simple components. It seems that returning null isn't correct if we want to replace the old component...

Reply:
It would be the complete migration package. Deprecate the old components and ask users to move over to this new pattern

Reply:
Thanks for commenting on deprecation, I changed the extension implementation to be compatible with the old components (instead of returning null, the extensions now render the original component), and I exported the extensions in place of the old components; this way, it is transparent to end consumers, they will automatically use the extensions behind the scenes when importing the old component. Sounds good?

Reply:
Oh interesting, yep I think that makes sense! 👍